### PR TITLE
Add site and slot public certificate

### PR DESCRIPTION
**Pull request by naveedaz**
Added schema for nested resource as a top level 
Added schema for nested resource as a child resource

### CODE REVIEW

**Comment by naveedaz** [comment]
@TianoMS , @daschult Please code review and merge.

File: schemas/2015-01-01/deploymentTemplate.json [hunk 1]
```diff
@@ -79,6 +79,7 @@
                   { "$ref": "http://schema.management.azure.com/schemas/2016-08-01/Microsoft.Web.json#/resourceDefinitions/sites_hybridConnectionNamespaces_relays" },
                   { "$ref": "http://schema.management.azure.com/schemas/2016-08-01/Microsoft.Web.json#/resourceDefinitions/sites_instances_deployments" },
                   { "$ref": "http://schema.management.azure.com/schemas/2016-08-01/Microsoft.Web.json#/resourceDefinitions/sites_premieraddons" },
+                  { "$ref": "http://schema.management.azure.com/schemas/2016-08-01/Microsoft.Web.json#/resourceDefinitions/sites_publicCertificates" },
```
Comment (TianoMS) [inline schemas/2015-01-01/deploymentTemplate.json]: >sites_publicCertificates [](start = 123, length = 24)

please add to 2014-04-01-preview/deploymentTemplate.json too.

File: schemas/2016-08-01/Microsoft.Web.json [hunk 2]
```diff
@@ -3037,6 +3145,40 @@
       ],
       "description": "Microsoft.Web/sites/premieraddons"
     },
+    "sites_publicCertificates_childResource": {
+      "type": "object",
+      "properties": {
```
Comment (TianoMS) [inline schemas/2016-08-01/Microsoft.Web.json]: >properties [](start = 7, length = 10)

Do you need "name" property on all the child resources?

File: schemas/2015-01-01/deploymentTemplate.json (hunk 1 above)
Comment (naveedaz) [inline schemas/2015-01-01/deploymentTemplate.json]: Fixed.

**Pull request merged**